mapit/views/areas.py

```diff
@@ -358,29 +363,13 @@ def areas_by_point(request, srid, x, y, bb=False, format='json'):
     method = 'box' if bb and bb != 'polygon' else 'polygon'
 
     args = query_args(request, format)
-    type = request.GET.get('type', '')
-
-    if type and method == 'polygon':
-        args = dict(("area__%s" % k, v) for k, v in args.items())
-        # So this is odd. It doesn't matter if you specify types, PostGIS will
-        # do the contains test on all the geometries matching the bounding-box
-        # index, even if it could be much quicker to filter some out first
-        # (ie. the EUR ones).
-        args['polygon__bbcontains'] = location
-        shapes = Geometry.objects.filter(**args).defer('polygon')
-        areas = []
-        for shape in shapes:
-            try:
-                areas.append(Area.objects.get(polygons__id=shape.id, polygons__polygon__contains=location))
-            except:
-                pass
+
+    if method == 'box':
+        args['polygons__polygon__bbcontains'] = location
     else:
-        if method == 'box':
-            args['polygons__polygon__bbcontains'] = location
-        else:
-            geoms = list(Geometry.objects.filter(polygon__contains=location).defer('polygon'))
-            args['polygons__in'] = geoms
-        areas = Area.objects.filter(**args)
+        geoms = list(Geometry.objects.filter(polygon__contains=location).defer('polygon'))
+        args['polygons__in'] = geoms
```

Comment:
To continue this, these two lines could now be `args['polygons__polygon__contains'] = location` I assume?